crowbar_framework/app/models/service_object.rb

```diff
@@ -1173,6 +1173,10 @@ def apply_role(role, inst, in_queue)
 
       nodes_in_batch = []
 
+      # get databag to remember potential removal of a role
+      databag = Chef::DataBag.load "barclamps/#{@bc_name}"
+      databag["crowbar"]["roles_to_remove"] = []
```

Comment:
It might have been my idea, so if it is, I apologize :-) But I think just adding an attribute in the data bag like this will break validation when we save a proposal that still has this attributes.

There might be a clever/hacky way to handle this, though (by putting the attributes under `elements`, maybe?).